flytekit/annotated/task.py

```diff
@@ -134,3 +138,49 @@ def wrapper(fn) -> PythonFunctionTask:
         return wrapper(_task_function)
     else:
         return wrapper
+
+
+class ReferenceTask(ReferenceEntity, PythonFunctionTask):
```

Comment:
I remember there was a circular import that I couldn't resolve when I tried moving the ReferenceTask into the task file itself.  Did you not run into this??

Reply:
No I didn't have a circular import issue.

Reply:
ooh i had previously tried to put it into python_function_task.